### PR TITLE
[REF] *: simplification of Record, KanbanRecord, HierarchyRenderer, and HierarchyCard

### PR DESCRIPTION
**Pull request by Polymorphe57**
Small refactoring of Record, KanbanRecord, HierarchyRenderer, and HierarchyCard components.

### CODE REVIEW

**Comment by Polymorphe57** [comment]
@robodoo rebase-ff

File: addons/web/static/src/model/record.js [hunk 1]
```diff
@@ -167,17 +167,21 @@ export class Record extends Component {
...
-            this.fields = this.props.fields;
+        const { activeFields, fieldNames, fields, resModel } = this.props;
+        if (!activeFields && !fieldNames) {
+            throw Error(`Record props should have either a "resModel" key or a "fieldNames" key`);
+        }
+        if (!fields && (!fieldNames || !resModel)) {
+            throw Error(
+                `Record props should have either a "fields" key or a "fieldNames" and a "resModel" key`
+            );
+        }
```
Comment (aab-odoo) [inline addons/web/static/src/model/record.js:178]: shouldn't this be done by the props validation mechanism? If we can't access other props in the validate function, it's maybe something missing in owl

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo delegate+

**Comment by Polymorphe57** [comment]
@robodoo r+ rebase-ff

File: addons/web/static/src/model/record.js (hunk 1 above)
Comment (aab-odoo) [reply addons/web/static/src/model/record.js:178]: I think you should create an owl issue

**Comment by tde-banana-odoo** [comment]
@aab-odoo @Polymorphe57 Thanks for simplifying our code. However you also broke the application.

When working in functional addons, consider asking for testing and review. Not the first time we have to fix "simplification" or "cleaning" done in submarine.

Cheers,

**Comment by aab-odoo** [comment]
@tde-banana-odoo Sorry for the inconvenience. That's the risk when doing changes in the framework. Hopefully that's also what the master branch is for. We also try to follow the odoo guidelines "break things move fast", and this one didn't look dangerous at first sight. Our mistake.

That being said, instead of a passive aggressive message, it would be nice to explain the issue so that we can fix it asap.

**Comment by tde-banana-odoo** [comment]
I don't see why you have to skip testing for changing 142 LOC. I never heard people screaming because some framework change did not land for an intermediate saas release. However I receive complaints when customers face tracebacks after 3 clicks. Not testing and hiding behind the stupid "move fast" motto is probably doing more harm than any passive aggressive message. We have to fight with what we have left I suppose.

Concerning the fix, it is under work in our team. If necessary we will ping you for review.

**Comment by aab-odoo** [comment]
No customer will face this issue as this is master, so that's fine. And it's not about merging this PR for the next freeze. It's about merging this PR s.t. other tasks depending on it can move forward.

We do various changes in the framework everyday. All of them have an impact on your applications. That's the joy of working in the framework. We can ping you to test each of our tasks if you want. But instead, we rely on tests. They often indicates when we break stuff. In this case, they didn't. I assume that the broken features wasn't tested.

**Comment by tde-banana-odoo** [comment]
``No customer will face this issue as this is master, so that's fine.``

And you block all tasks, testing, review, ... in broken applications, as we have to stop development, understand why it is suddenly broken, find a fix, merge it, then resume development. You gain one hour, we lose one day. You are not alone in Odoo R&D. So yes, you have to test outside of your test suite.

**Comment by aab-odoo** [comment]
I still don't know what the issue is...

File: addons/web/static/src/model/record.js [hunk 2]
```diff
@@ -167,17 +167,21 @@ export class Record extends Component {
         "onWillSaveRecord?",
     ];
     setup() {
-        if (this.props.fields) {
-            this.fields = this.props.fields;
+        const { activeFields, fieldNames, fields, resModel } = this.props;
+        if (!activeFields && !fieldNames) {
+            throw Error(`Record props should have either a "resModel" key or a "fieldNames" key`);
```
Comment (geflx) [inline addons/web/static/src/model/record.js:172]: Hello :wave:
I think there is a typo in `resModel`, should it be `activeFields`?
I noticed it while debugging the error during a task. ^^

**Pull request closed without merging**